---
title: "Linkage Map Construction"
author: "Matthew Gibson"
date: "June 15, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Load packages and set wd
```{r}
library(qtl)
library(ASMap)

pcDir <- "c:/Users/matth/Documents/bin/jaltomataQTL/mapping_3"
setwd(pcDir)
```


### Load in genotype data from Stacks
```{r}
jal_cross <- read.cross("csv",pcDir, "batch_1.genotypes_20.filtered.loc", genotypes=c("b", "h"), alleles=c("A", "B"), na.strings=c("-", "NA"), estimate.map=F,crosstype = "bc")
```

### Visualize and filted markers.
```{r}

par(mfrow=c(1,2), las=1)
plot(ntyped(jal_cross), ylab="No. typed markers", main="No. genotypes by individual")
plot(ntyped(jal_cross, "mar"), ylab="No. typed individuals",main ="No. genotypes by marker")

nt.bymar <- ntyped(jal_cross,"mar")
todrop <- names(nt.bymar[nt.bymar < 80])
jal_cross <- drop.markers(jal_cross, todrop)
summary(jal_cross)
```

### Diagnostic plots and removing distorted markers.
```{r}
profileMark(jal_cross, stat.type=c("seg.dist", "prop", "miss"), crit.val = "bonf", layout=c(1,4), type="l", cex=0.5)
mm <- statMark(jal_cross, stat.type = "marker")$marker$AB
mapBC3 <- drop.markers(jal_cross, c(markernames(jal_cross)[mm > 0.8],markernames(jal_cross)[mm < 0.2]))
```

### Initial map construction
```{r}
mapJal0 <- mstmap(mapBC3, bychr = F, dist.fun = "kosambi", trace = TRUE, p.value = 1e-35, id="index")
nmar(mapJal0)
chrlen(mapJal0)
```

### Looking at linkage of smaller groups to the larger ones. 
```{r}
#The following chunk was repeated for every small LG produced in intitial build
lod <- pull.rf(mapJal0, what="lod")
mn <- markernames(mapJal0, chr='L.10')
plot(lod, mn[1], bandcol="gray70", alternate.chrid=T)
```

### Removing small groups with no apparent linkage to other groups, merging those that are. 
```{r}
mapJal_lightMerge <- drop.markers(mapJal0, markernames(mapJal0, c('L.10', 'L.14', 'L.15', 'L.16', 'L.18', 'L.23', 'L.24', 'L.25', 'L.26', 'L.28', 'L.34')))
mapJal_lightMerge <- mergeCross(mapJal_lightMerge, merge= list('L.8'=c('L.8','L.12', 'L.31')))
mapJal_lightMerge <- mergeCross(mapJal_lightMerge, merge= list('L.17'=c('L.17','L.13', 'L.20')))
mapJal_lightMerge <- mergeCross(mapJal_lightMerge, merge= list('L.7'=c('L.7','L.27', 'L.21', 'L.33')))
mapJal_lightMerge <- mergeCross(mapJal_lightMerge, merge= list('L.5'=c('L.5','L.22')))
mapJal_lightMerge <- mergeCross(mapJal_lightMerge, merge= list('L.9'=c('L.9','L.29', 'L.30')))
mapJal_lightMerge <- mergeCross(mapJal_lightMerge, merge= list('L.2'=c('L.2', 'L.32')))
```

### Re-order based on updated groups.
```{r}
mapJal1  <- mstmap(mapJal_lightMerge, bychr = T, dist.fun = "kosambi", trace = TRUE, p.value = .000001, id="index")
nmar(mapJal1)
chrlen(mapJal1)
```

### Diagnostic plots and long length of LGs point to genotpying errors. Beyond theoretical length limit. 
```{r}
profileMark(mapJal1.1, stat.type = c("seg.dist", "prop", "dxo", "recomb"),layout = c(1, 5), type = "l")
```

### Write map to file and run through `correctGenotypes.py` which removes markers that have genotypes which consistently disagree with its neighbors
```{r}
#write.cross(mapJal1, format="csvr")
```

### Reload cross
```{r}
jal_cross <- read.cross("csv",pcDir, "filtered_mapJal1.csv", genotypes=c("A", "H"), alleles=c("A", "B"), na.strings=c("-", "NA"), estimate.map=F,crosstype = "bc")
```

### Recalculate map, maintain LGs
```{r}
mapJal3 <- mstmap(jal_cross, bychr = T, dist.fun = "kosambi", trace = TRUE, p.value = 2, id="index")
nmar(mapJal3)
profileMark(mapJal3, stat.type = c("seg.dist", "prop", "dxo", "recomb"),layout = c(1, 5), type = "l")

#write.cross(mapJal3, format="csvr")
```

### More stringent threshold. 
```{r}
mapJal4 <- read.cross("csv",pcDir, "mapJal4.csv", genotypes=c("A", "H"), alleles=c("A", "B"), na.strings=c("-", "NA"), estimate.map=F,crosstype = "bc")
mapJal4 <- mstmap(mapJal4, bychr = T, dist.fun = "kosambi", trace = TRUE, p.value = 2, id="index")
nmar(mapJal4)
plot.map(mapJal4)
mapJal4 <- jittermap(mapJal4, amount=.3)
#write.cross(mapJal4, format="csvr")
```

### Map is still a little long, with smaller marker set now we can probably use `ripple()` in R/QTL.
```{r}
#Was done for all chromosomes
rip <- ripple(mapJal4, chr="L.1", window=6)
summary(rip)
#mapJal4 <- switch.order(mapJal4, rip[2,])

#write.cross(mapJal4, format="csvr")
```

# This is the correct map
```{r}
finalMap <- read.cross("csv",pcDir, "final_map_6_15_t.csv", genotypes=c("A", "H"), alleles=c("A", "B"), na.strings=c("-", "NA"), estimate.map=F,crosstype = "bc")

plot.map(finalMap)
